### DISCUSSION TITLE
Opening duration of large files

### DISCUSSION THREAD

**Discussion by AlexanderJuestel** [geopandas/geopandas#3136] [Q&A]
Hello, 

I am working with rasters and vector datasets with sizes way beyond 1 GB (for many probably still not very big). However, while rasters just take a few seconds at most to get opened with rasterio, vector datasets (shapefiles) may take up to hours to open using GeoPandas. 

The vector datasets may contain dozens of columns and geometries with many vertices. 

Is there a way to accelerate this process?

**Comment by martinfleis** [comment]
The fastest option would be to use `pyogrio` engine with the arrow stream enabled. It means you have to ensure that you have `pyogrio` and `pyarrow` packages installed and then you can do


```py
gpd.read_file(path, engine="pyogrio", use_arrow=True)
```

Another option is to try reading the file in parallel with `dask-geopandas`.

Note that the comparison with raster is not very relevant as 1) you may be lazily reading it, not copying to memory, 2) mapping of data from raster to memory is straightforward while deserialisation of geometries and conversion to shapely objects takes time.

**Comment by theroggy** [comment]
In addition to what Martin wrote... Shapefile is an old file format with quite some disadvantages ... and also regarding performance it isn't the most optimised format. So if wanted/needed, that is a possible additional source of improvement.

E.g. reading a Geopackage file is in general already faster, but with `use_arrow=True`, under the hood different parts of the file will be read in parallel to further improve performance.

For illustration, a comparison between reading a file with 3.2 million polygons:
* as Shapefile (3.12 GB): 41 s
* as Geopackage (1.4 GB): 8.3 s

```
from datetime import datetime
from pathlib import Path

import geopandas as gpd

gpkg_path = Path("C:/temp/lds-nz-building-outlines/nz-building-outlines.gpkg")
shp_path = gpkg_path.with_suffix(".shp")

if not shp_path.exists():
    gfo.copy_layer(gpkg_path, shp_path)

start = datetime.now()
_ = gpd.read_file(gpkg_path, engine="pyogrio", use_arrow=True)
print(f"read gpkg took {datetime.now()-start}")

start = datetime.now()
_ = gpd.read_file(shp_path, engine="pyogrio", use_arrow=True)
print(f"read shp took {datetime.now()-start}")
```
